src/tests/performance/prio_test.rs

```diff
@@ -0,0 +1,313 @@
+    // │       1 │               400 │        300 │          1 │          9 │              1 │         100.00 │       0.82 │
+    // └─────────┴───────────────────┴────────────┴────────────┴────────────┴────────────────┴────────────────┴────────────┘
+    // *NOTE* run with `cargo test --release  tests::performance::prio -- --nocapture`
+    // Before running test generate dummy devices with different ips using toos/ips.py
+
+    let synth_counts = vec![1, 50, 100, 200, 300, 400];
+
+    let mut table = RequestsTable::default();
+
+    for synth_count in synth_counts {
+        // synth_count malicious tasks plus one normal synth_node
+        let barrier = Arc::new(Barrier::new(synth_count + 1));
+
+        let target = TempDir::new().expect("couldn't create a temporary directory");
+        let mut node = Node::builder()
+            .build(target.path())
+            .expect("unable to build the node");
+        node.start().await;
+
+        let node_addr = node.net_addr().expect("network address not found");
+
+        let mut synth_sockets = Vec::with_capacity(synth_count + 1);
+        let mut ips = IPS.to_vec();
+
+        for _ in 0..synth_count + 1 {
+            // If there is address for our thread in the pool we can use it.
+            // Otherwise we'll not set bound_addr and use local IP addr (127.0.0.1).
+            let ip = ips.pop().unwrap_or("127.0.0.1");
+
+            let ip = SocketAddr::new(IpAddr::V4(Ipv4Addr::from_str(ip).unwrap()), 0);
+            let socket = TcpSocket::new_v4().unwrap();
+
+            // Make sure we can reuse the address and port
+            socket.set_reuseaddr(true).unwrap();
+            socket.set_reuseport(true).unwrap();
+
+            socket.bind(ip).expect("unable to bind to socket");
+            synth_sockets.push(socket);
+        }
+
+        // setup metrics recorder
+        let test_metrics = TestMetrics::default();
+        // clear metrics and register metrics
+        metrics::register_histogram!(METRIC_LATENCY);
+
+        let mut synth_handles = JoinSet::new();
+        let test_start = tokio::time::Instant::now();
+
+        let arc_barrier = barrier.clone();
+        synth_handles.spawn(simulate_normal_peer(
+            node_addr,
+            synth_sockets.pop().unwrap(),
+            arc_barrier,
+        ));
+
+        for socket in synth_sockets {
+            let arc_barrier = barrier.clone();
+            synth_handles.spawn(simulate_malicious_peer(node_addr, socket, arc_barrier));
+        }
+
+        // wait for peers to complete
+        while (synth_handles.join_next().await).is_some() {}
+
+        let time_taken_secs = test_start.elapsed().as_secs_f64();
+
+        let snapshot = test_metrics.take_snapshot();
+        if let Some(latencies) = snapshot.construct_histogram(METRIC_LATENCY) {
+            if latencies.entries() >= 1 {
+                // add stats to table display
+                table.add_row(RequestStats::new(

```

Comment:
```suggestion
        if let Some(latencies) = snapshot.construct_histogram(METRIC_LATENCY) && if latencies.entries() >= 1 {
            // add stats to table display
            table.add_row(RequestStats::new(

```

Reply:
`let expressions in this position are unstable` - that's what I get after removing `if` otherwise it won't compile. 
Seems to be some testing feature on my `rustc 1.67.0-nightly (70f8737b2 2022-11-23)`